### DISCUSSION TITLE
Not able to access packages with Emacs 27.2

### DISCUSSION THREAD

**Discussion by pedro-w** [Homebrew/discussions#1475] [Everyday usage] [answered]
I use the emacs package manager to download elisp packages and recently (in the last month or so) it's stopped updating. Now it just hangs for a long time, and in the messages window I see:
```plain
error in process sentinel: Error retrieving: https://elpa.gnu.org/packages/archive-contents (error connection-failed "connect" :host "elpa.gnu.org" :service 443) [2 times]
Package refresh done
error in process sentinel: Error retrieving: https://melpa.org/packages/archive-contents (error connection-failed "connect" :host "melpa.org" :service 443) [2 times]
``` 
I can fetch the contents of `https://melpa.org/packages/archive-contents` from the command line but I can't using eww from within emacs.
Has anyone else seen this? It looks like a network connection thing. I don't know if it is something specific to my installation, or homebrew or emacs in general.

Specs: Mac running Mojave 10.14.6
```plain
$ brew info homebrew/cask/emacs
emacs: 27.2-2
https://emacsformacosx.com/
/usr/local/Caskroom/emacs/27.2-2 (5 files, 269.3KB)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/emacs.rb
```
Nothing relevant from `brew doctor`
Relevant part of `.emacs`
```elisp
(package-initialize)
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)
```
I've also tried starting emacs with `-q` and starting the package manager manually from the scratch window.

**Comment by carlocab** [comment, marked as answer]
Almost certain it's not a Homebrew issue. You can check for yourself by uninstalling the cask then installing Emacs using the installer at https://emacsformacosx.com/.

**Reply by pedro-w** [reply]
You're right, I uninstalled / installed as you suggest and the problem remained. So I rolled up my sleeves and did a bit more googling.
I've fixed it now (I hope!) by adding 
```elisp
(setq gnutls-algorithm-priority "NORMAL:-VERS-TLS1.3")
```
to my `.emacs` - most sources say this shouldn't be needed after 26.2 but I found [someone saying](https://emacs.stackexchange.com/questions/54226/accessing-https-sites-with-mac-emacs-26-3#comment84668_54247) it might be needed still on MacOS, so I gave it a try.
I'm back on the homebrew emacs now.
